intents/oversights/simpsons_paradox.py

```diff
@@ -0,0 +1,292 @@
+        # is not in the grouping list initially
+        if column not in dimension_list:
+            new_grouping_columns = dimension_list.copy()
+            new_grouping_columns.append(column)
+            new_grouping_columns.append(slice_compare_column)
+
+            new_required_columns = new_grouping_columns.copy()
+            new_required_columns.append(metric)
+            new_cropped_table = aspects.crop_other_columns(table, 
+                                                           new_required_columns)
+
+            # result table after adding the new column in the grouping list.
+            new_result_table = aspects.group_by(new_cropped_table,
+                                                new_grouping_columns,
+                                                summary_operator)
+            
+            # it will return the debiasing suggestion after comparing the
+            # initial result table and new result table.
+            new_suggestion =  _check_simpsons_paradox(initial_result_table,
+                                                      new_result_table, 
+                                                      column, slice1,
+                                                      dimensions_len)
+            if new_suggestion != None:
+                suggestions.append(new_suggestion)
+    return suggestions
+
+def _check_simpsons_paradox(initial_result_table, new_result_table, new_added_column,
+                                                              slice1, dimensions_len):
+    """
+    Args:
+        initial_result_table: Type-pandas.dataframe
+            It has the contents of the csv file
+        new_result_table: Type-pandas.dataframe
+            It has the contents of the csv file
+        new_added_column: Type-string
+            The new column added in the initial grouping column.
+        slice1: Type-string
+            the first slice by which we do comparision.
+        dimensions_len: Type-integer
+            size of the initial gfrouping dimensions.
+
+    Returns:
+        return a dictionary where each dictionary represent a debiasing 
+               suggestion according to the new column.
+    """
+    initial_num_rows = initial_result_table.shape[0]
+    new_num_rows = new_result_table.shape[0]
+
+    # changing dataframe to list of list to do next set of operation
+    initial_table_matrix = initial_result_table.values.tolist()
+    new_table_matrix = new_result_table.values.tolist()
+
+    initial_row_i = 0
+    new_row_i = 0
+
+    suggestion_row_list = []
+
+    """ dominant percentage is the percentage of pairs where 
+        (first value - second value) is positive."""
+
+    while initial_row_i < initial_num_rows:
+
+        # check if it is the last row or there is only one slice for this group.
+        if initial_row_i == initial_num_rows - 1 or initial_table_matrix[initial_row_i][:dimensions_len] != initial_table_matrix[initial_row_i + 1][:dimensions_len]:
```

Comment:
There seems to be some code / logic duplication between the if and the else cases. Could we have the same code but with different values computed for initial_difference?

Reply:
removed the code duplication